frontend/src/test/java/com/google/sps/InMemoryTripModelStorageTest.java

```diff
@@ -0,0 +1,266 @@
+package com.google.sps;
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.common.truth.Truth8.assertThat;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import java.util.List;
+import com.google.tripmeout.frontend.storage.InMemoryTripModelStorage;
+import com.google.tripmeout.frontend.TripModel;
+import com.google.tripmeout.frontend.error.TripAlreadyExistsException;
+import com.google.tripmeout.frontend.error.TripNotFoundException;
+
+
+public class InMemoryTripModelStorageTest {
+  public static final TripModel LONDON_TRIP1 = TripModel.builder()
+      .setId("a")
+      .setName("London Trip")
+      .setUserId("person 1")
+      .setLocationLat(32.333)
+      .setLocationLong(52.1)
+      .build();
+
+  public static final TripModel LONDON_TRIP2 = TripModel.builder()
+      .setId("b")
+      .setName("London Trip")
+      .setUserId("person 2")
+      .setLocationLat(32.333)
+      .setLocationLong(52.1)
+      .build();
+
+  public static final TripModel LONDON_TRIP1_2022 = TripModel.builder()
+      .setId("c")
+      .setName("London Trip 2022")
+      .setUserId("person 1")
+      .setLocationLat(32.333)
+      .setLocationLong(52.1)
+      .build();
+
+  public static final TripModel TOKYO_TRIP1 = TripModel.builder()
+      .setId("d")
+      .setName("Tokyo Trip")
+      .setUserId("person 1")
+      .setLocationLat(32.333)
+      .setLocationLong(147.1112)
+      .build();
+
+  public static final TripModel LONDON_TRIP_DUPLICATE_ID = TripModel.builder()
+      .setId("a")
+      .setName("London Trip")
+      .setUserId("person 2")
+      .setLocationLat(32.333)
+      .setLocationLong(52.1)
+      .build();
+
+
+  /**
+   * test that an exception is thrown when adding a TripModel that has already 
+   * been added to storage
+   */
+  @Test
+  public void addTrip_tripAlreadyAdded_throwsException() 
+      throws TripAlreadyExistsException {
+    InMemoryTripModelStorage storage = new InMemoryTripModelStorage();
+    storage.addTrip(LONDON_TRIP1);
+    Assert.assertThrows(TripAlreadyExistsException.class, () -> 
+        storage.addTrip(LONDON_TRIP1));
+    
+  }
+
+  /**
+   * test that an exception is thrown when adding a TripModel when there is 
+   * already a TripModel object in storage with the same tripId
+   */
+  @Test
+  public void addTrip_tripWithSameIdAlreadyAdded_throwsException() 
+      throws TripAlreadyExistsException {
+    InMemoryTripModelStorage storage = new InMemoryTripModelStorage();
+    storage.addTrip(LONDON_TRIP1);
+    Assert.assertThrows(TripAlreadyExistsException.class, () -> 
+        storage.addTrip(LONDON_TRIP_DUPLICATE_ID));
+    
+  }
+
+  /**
+   * test that adding a different TripModel object does not cause an exception
+   * to be thrown
+   */
+  @Test
+  public void addTrip_tripNotInStorage_exceptionNotThrown() 
```

Comment:
This case may just be adding more test burden without being very revealing. You generally want each test case to end in an assertion (ideally the _only_ assertion made).

Here, it seems like the semantic behavior we're testing is whether the storage system correctly supports multiple trips with the same name. If so, then we should be testing that those values can actually be retrieved and match their expected values. (If that's not what you're trying to test, then figure out how to spell the test case with storage interactions that can be asserted.)